### PR TITLE
ci: compute size diff between PR and last release

### PR DESCRIPTION
**Pull request by Lou1415926**
instead of mainline vs. last release

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Comment by dannyrandall** [comment]
This workflow is making me wonder if we can use a different event: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/ 🤔

**Review by efekarakus — APPROVED** [review]

**Review by huanjani — APPROVED** [review]

**Comment by Lou1415926** [comment]
I updated this PR so that we now have two workflows:
1. `Compute bin sizes` is triggered by `pull_request` event, which is a read-only trigger. This workflow computes the sizes of old/new binaries, and upload the results as artifacts.
2. `Comment on PR` is triggered by the completion of `Compute bin sizes`. It reads the results that were uploaded as artifacts, and comment on the PR.

I already have two approvals for the previous version, but this change is too large to assume that the approvals are still in effect 👀  Can I please have two more approvals before I can remove the label?

File: .github/workflows/ci_size_computer.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          echo "MAC_ARM_KIB=$(du -k ./bin/local/copilot-darwin-arm64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "LINUX_AMD_KIB=$(du -k ./bin/local/copilot-linux-amd64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "LINUX_ARM_KIB=$(du -k ./bin/local/copilot-linux-arm64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "WINDOWS_AMD_KIB=$(du -k ./bin/local/copilot.exe | awk '{ print $1}')" >> $GITHUB_OUTPUT
+
+      - name: Save sizes and PR number
+        run: |
+          mkdir -p artifacts/old artifacts/new
+          sizes_json="{
+            \"macOS\": {
```
Comment (efekarakus) [inline .github/workflows/ci_size_computer.yml:64]: lol, this must have been fun to write

**Review by efekarakus — APPROVED** [review]
Really cool! Thanks Wanxian!

File: .github/workflows/ci_size_computer.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          make release
+          echo "MAC_AMD_KIB=$(du -k ./bin/local/copilot-darwin-amd64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "MAC_ARM_KIB=$(du -k ./bin/local/copilot-darwin-arm64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "LINUX_AMD_KIB=$(du -k ./bin/local/copilot-linux-amd64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "LINUX_ARM_KIB=$(du -k ./bin/local/copilot-linux-arm64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "WINDOWS_AMD_KIB=$(du -k ./bin/local/copilot.exe | awk '{ print $1}')" >> $GITHUB_OUTPUT
+
+      - name: Save sizes and PR number
+        run: |
+          mkdir -p artifacts/old artifacts/new
```
Comment (dannyrandall) [inline .github/workflows/ci_size_computer.yml]: why do we need this line, and not just `mkdir artifacts`?

**Review by dannyrandall — APPROVED** [review]
just a question, otherwise looks great!🚀

File: .github/workflows/ci_size_computer.yml (hunk 2 above)
Comment (Lou1415926) [reply .github/workflows/ci_size_computer.yml]: good catch! I changed the implementation and forgot to remove this line

**Pull request merged**